distribution/tools/launchers/src/test/java/org/elasticsearch/tools/launchers/JvmOptionsParserTests.java

```diff
@@ -149,6 +156,108 @@ public void testComplexOptions() throws IOException {
         }
     }
 
+    public void testMissingRootJvmOptions() throws IOException, JvmOptionsParser.JvmOptionsFileParserException {
+        final Path config = newTempDir();
+        try {
+            final JvmOptionsParser parser = new JvmOptionsParser();
+            parser.readJvmOptionsFiles(config);
+            fail();
```

Comment:
Should we provide a message for this assertion such as "Expected NoSuchFileException but no exception was thrown"?

Reply:
A pattern I've sometimes used is to capture any exception thrown, then assert that it's:

   * Not null
   * Has the right class
   * Has the right message

Up to you if you think that's overkill, but at least if the test fails then it'll tell you exactly why.

Reply:
What we're doing here is effectively doing the first two items of your list. I'm so-so on testing the message in most cases. FWIW, what you describe is trivial [using Spock](https://github.com/elastic/elasticsearch-infra/blob/42df194abc6b31837fd425b262050eab7f4c5020/homer/src/test/groovy/co/elastic/homer/slack/SlackApiClientSpec.groovy#L79-L81) 😉

Reply:
As @mark-vieira mentions, that's not needed here because we have an explicit catch for the expected exception type which ensures non-nullness, and also that it's the right type. Asserting on the exception message here is not needed because it's going to come from the operating system, it's not an exception message that we control.

There are cases where the three checks are appropriate. For example, when asserting that a listener is called back with the proper exception, asserting non-null and the type are good. The message can be a valuable assertion when we need to distinguish between two reasons that a given exception type might arise (e.g., `IllegalArgumentException`). It a lot of cases, asserting on an exception message is a smell, especially in production code, and in those cases I prefer that make dedicated exception types (we've done this a few times in CCR, when an exception is going to cross the wire, for example).